src/modules/service-catalog/useServiceCatalogItem.tsx

```diff
@@ -0,0 +1,28 @@
+import { useEffect, useState } from "react";
+import type { ServiceCatalogItem } from "./data-types/ServiceCatalogItem";
+
+export function useServiceCatalogItem(
+  serviceItemId: number
+): ServiceCatalogItem | undefined {
+  const [serviceCatalogItem, setServiceCatalogItem] = useState<
+    ServiceCatalogItem | undefined
+  >();
+
+  useEffect(() => {
+    const fetchServiceCatalogItem = async () => {
+      try {
+        const response = await fetch(
+          `/api/v2/help_center/service_catalog/items/${serviceItemId}`
+        );
+        const data = await response.json();
+        setServiceCatalogItem(data.service_catalog_item);
+      } catch (error) {
+        console.error("Error fetching service catalog item:", error);
```

Comment:
Unless I'm missing something here, looks like we are covering only the request happy path, ie when fetch returns a 200 response. What would happen if we get `404` or `503`? 

Shouldn't we check for `!response.ok or !response.ok` and provide some error handling?

@luis-almeida

Reply:
That's true I missed that. Can we check `response.ok` and throw an error so it falls in `catch`?